app/routers/legal_docs_generator/legal_docs.py

```
@@ -0,0 +1,29 @@
+from app.routers.legal_docs_generator.deepseek import deepseek_stream_response
+import httpx
+import asyncio
+
+router = APIRouter()
+
+DEEPSEEK_API_URL = "http://localhost:8000/deepseek"  # Internal DeepSeek API URL
+
+async def fetch_deepseek_response(request_data):
+    """Send request to DeepSeek and stream the response"""
+    async with httpx.AsyncClient() as client:
+        async with client.stream("POST", DEEPSEEK_API_URL, json=request_data) as response:
```

Ruff (0.8.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)